Configurations.md

```diff
@@ -2362,9 +2362,63 @@ Require a specific version of rustfmt. If you want to make sure that the
 specific version of rustfmt is used in your CI, use this option.
 
 - **Default value**: `CARGO_PKG_VERSION`
-- **Possible values**: any published version (e.g. `"0.3.8"`)
+- **Possible values**: `semver` compliant values, such as defined on [semver.org](https://semver.org/).
 - **Stable**: No (tracking issue: [#3386](https://github.com/rust-lang/rustfmt/issues/3386))
 
+#### Match on exact version:
+
+```toml
+required_version="1.0.0"
+```
+
+#### Higher or equal to:
+
+```toml
+required_version=">=1.0.0"
+```
+
+#### Lower or equal to:
+
+```toml
+required_version="<=1.0.0"
+```
+
+#### New minor or patch versions:
+
+```toml
+required_version="^1.0.0"
+```
+
+#### New patch versions:
+
+```toml
+required_version="~1.0.0"
+```
+
+#### Wildcard:
+
+```toml
+required_version="*" # matches any version.
+required_version="1.*" # matches any version with the same major version
+required_version="1.0.*" # matches any version with the same major and minor version
+```
+
+#### Multiple versions to match:
+
+Except by `*`, any of the semver operators can be combined, being split with commas. The comparison is done using `&&` operator.
+
+`*` can't be used with other comparators.
+
+Since `*` has range restrictions, any comparator will override the wildcard operator. When `*` is used alone, the comparison always fails, as that's an invalid config. `*, <1.0.0` for example is invalid.
+
+Versions can't contradict themselves, otherwise they'll always fail. Some examples are `"1.*, <2.0.0"` and `1.0.*, <2.0.0`, since both versions can't be true at the same time.
+
+By now, `&&` (using comma `,`) is the only way to declare `required_version`. `||` Operator is not valid. Valid examples are `1.0.0, 1.1.0` and `1.0.0, 1.1.0, 1.1.*` but not `1.0.0 || 1.1.0` or `1.0.0, 2.0.0`.
```

Comment:
Supporting `||` matters to us, or can we keep only the comma operator?

Reply:
Sure, was too into this, mb. 

So, the `semver` crate works so that you can send multiple versions, which should be compatible. It is like the use of an "and" operator. So your version can be something like `^1.0.0, 1.0.*`. But can't be `^1.0.0, ^2.0.0`.

My question is whether it's important to us to support multiple versions, which are not necessarily always true. 

Are we okay with multiple versions translating from `^1.0.0, ^2.0.0` being (matches version a AND version b)? or do we want this to be "matches version a OR version b".

Let me know if it's clearer now.

Reply:
That's clearer, thanks. Do other tools with semver functionality support `OR` with `||`? Does the `semver` crate support that functionality? If not then I don't think it's necessary. That said, I don't want to increase the scope of the current PR so I'd still want to hold off on `||` until a follow up PR.

Reply:
I don't think so unless they're implementing it manually. 

Agree with you that we can go as it is and implement a follow-up PR.

Reply:
Given that we're not going to support `OR` logic with `||` I'd prefer it if we removed that language from the docs that you wrote.

Also I'm having a hard time understanding why the following example is contradictory:

> Versions can't contradict themselves, otherwise they'll always fail. Some examples are `"1.*, <2.0.0"` and `1.0.*, <2.0.0`, since both versions can't be true at the same time.

Why can't both be true? Isn't the first example communicating that we'll accept any `v1` and the second is saying we want any `v1.0.*`? The `<2.0.0` feels redundant, but I don't think it's contradictory. Maybe an example like `<1.5.0, >1.10.*` would make more sense for something that's contradictory?

Reply:
Yeah actually the example is wrong; what I meant is `1.*, >2.0.0`

Reply:
Fixed the docs and included your example.